IA/IA_upload.py

```diff
@@ -0,0 +1,107 @@
+import os
+import argparse
+import requests
+import boto
+from boto.s3.connection import OrdinaryCallingFormat
+from boto.s3.multipart import MultiPartUpload
+import asyncio
+import xmltodict
+from io import BytesIO
+from settings import (
+    CHUNK_SIZE,
+    OSF_COLLECTION_NAME,
+    IA_ACCESS_KEY,
+    IA_SECRET_KEY,
+)
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+
+class IAException(Exception):
+    pass
+
+
+def mp_from_ids(mp_id: str, mp_keyname: str, bucket: str) -> MultiPartUpload:
+    mp = MultiPartUpload(bucket)
+    mp.key_name = mp_keyname
+    mp.id = mp_id
+    return mp
+
+
+async def gather_and_upload(bucket_name: str, parent: str):
+    '''
+    This script traverses through a directory uploading everything in it to Internet Archive.
+    '''
+
```

Comment:
The bagit library provides a `bag.is_valid()` function. This is very much a nice to have, but it would be great to ensure that what we are uploading to IA is a valid Bag. Plus its just two lines of code.

Reply:
I'm actually thinking of adding this to the bag and tag ticket instead.

Reply:
That sounds good too. I was pondering where this would fit better after I made this comment 👍